infra/build/functions/sync/main.py

```diff
@@ -35,41 +41,155 @@ class GitAuth(ndb.Model):
+    except exceptions.GoogleAPICallError as error:
+      print(error.message)
+
+    except exceptions.RetryError as error:
+      print(error.message)
+
+    except ValueError:
+      print("Incorrect parameters passed")
+
+
+def update_scheduler(cloud_scheduler_client, project, schedule):
+  """Updates schedule in case schedule was changed."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project.name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': bytes(project.name, 'utf-8')
+      },
+      'schedule': project.schedule
+  }
+
+  update_mask = {'schedule': schedule}
+
+  try:
+    cloud_scheduler_client.update(job, update_mask)
+
+  except exceptions.GoogleAPICallError as error:
+    print(error.message)
+
+  except exceptions.RetryError as error:
+    print(error.message)
+
+  except ValueError:
+    print("Incorrect parameters passed")
+
+
+def sync_projects(cloud_scheduler_client, projects):
   """Sync projects with cloud datastore."""
   project_query = Project.query()
   projects_to_remove = [
       project.key for project in project_query if project.name not in projects
   ]
+  schedulers_to_remove = [
+      project.name for project in project_query if project.name not in projects
+  ]
 
   ndb.delete_multi(projects_to_remove)
+  delete_schedulers(cloud_scheduler_client, schedulers_to_remove)
 
   existing_projects = {project.name for project in project_query}
 
   new_projects = [
-      Project(name=project)
-      for project in projects
-      if project not in existing_projects
+      Project(name=project_name, schedule=projects[project_name])
+      for project_name in projects
+      if project_name not in existing_projects
   ]
+
+  for project_name in projects:
+    if project_name not in existing_projects:
+      create_scheduler(cloud_scheduler_client, project_name,
+                       projects[project_name])
+
   ndb.put_multi(new_projects)
 
+  for project in project_query:
+    if project.name in projects and project.schedule != projects[project.name]:
+      update_scheduler(cloud_scheduler_client, project, projects[project.name])
+      project.schedule = projects[project.name]
+      project.put()
+
 
-def _has_docker_file(repo, project_path):
+def _has_docker_file(project_contents):
   """Checks if project has a Dockerfile."""
-  return any(content_file.name == 'Dockerfile'
-             for content_file in repo.get_contents(project_path))
+  return any(
+      content_file.name == 'Dockerfile' for content_file in project_contents)
+
+
+def get_schedule(project_contents):
+  """Checks for schedule parameter in yaml file else uses DEFAULT_SCHEDULE."""
+  for content_file in project_contents:
+    if content_file.name == 'project.yaml':
```

Comment:
prefer bailing early rather than introducing another indentation level.

i.e.

```python
if content_file.name != 'project.yaml':
  continue
```